clio-client/src/test/scala/org/broadinstitute/clio/client/webclient/ClioWebClientSpec.scala

```diff
@@ -176,6 +177,44 @@ class ClioWebClientSpec extends BaseClientSpec with AsyncMockFactory {
     }
   }
 
+  it should "submit arbitrary JSON as a raw query from a JSON input file" in {
+    val index = ModelMockIndex()
+
+    val jsonInput = Map("this" -> "is valid json").asJson
+    val expectedSegments =
+      Seq(
+        ApiConstants.apiString,
+        "v1",
+        index.urlSegment,
+        ApiConstants.rawQueryString
+      )
+
+    val flow = Flow[HttpRequest].map { req =>
+      req.uri.path.toString should be(expectedSegments.mkString("/", "/", ""))
+      req.entity should be(
+        HttpEntity(
+          ContentTypes.`application/json`,
+          jsonInput.pretty(defaultPrinter)
+        )
+      )
+      jsonResponse(jsonInput)
+    }
+
+    val generator = mock[CredentialsGenerator]
+    (generator.generateCredentials _).expects().returning(OAuth2BearerToken("fake"))
+
+    val client = new ClioWebClient(flow, timeout, 0, generator)
+
+    (for {
+      tempFile <- File.temporaryFile()
+      _ = tempFile.writeText(jsonInput.pretty(defaultPrinter))
+    } yield
+      client
+        .jsonFileQuery(index)(tempFile)
+        .runWith(Sink.head)
+        .map(_ => succeed)).get
```

Comment:
`for`-`yield` wrapped with parens is ugly... since there's only one `flatMap`, I'd just be explicit with using it. I think the call to `.get` will properly resolve without extra parens that way.